Gemfile

```diff
@@ -32,7 +32,7 @@ gem 'oj'
 # use jquery ui to interact with frontend elements
 gem 'jquery-ui-rails'
 # Use ActiveModel has_secure_password
-# gem 'bcrypt', '~> 3.1.7'
+gem 'bcrypt', '~> 3.1.7'
```

Comment:
Since you're doing your own key encryption, it might be worth it to look at the [scrypt](https://rubygems.org/gems/scrypt).

<img src="https://d262ilb51hltx0.cloudfront.net/max/800/1*QdbniDuZiiF1N7ArNJChOA.png"/>